### DISCUSSION TITLE
SharedPreferences usage in qlevarRouter Controller file

### DISCUSSION THREAD

**Discussion by guruprakashIT** [SchabanBo/qlevar_router#15] [Q&A]
Is there any option available to use Sharedpreferences in QlevarRouter Controller file @SchabanBo ,. 

I need to dynamically pass value to router file?

thanks!

**Comment by SchabanBo** [comment]
I'm sorry I didn't quite understand your question.
What do you mean with router file?
Can you give an example of what you need?

**Reply by guruprakashIT** [reply]
localhost:8080/{dynamic name}/dashboard
dynamic name eg. domain1,domain45...etc
i have that string in shared preference . how do I pass this string to url

**Reply by SchabanBo** [reply]
you could use Middlewares
```dart
 QRoute(
    path: '/:domain/dashboard',
     middleware: [
      QMiddlewareBuilder(redirectGuardFunc: (s) async {
          final doamins = await getDoaminsFromSharedPreferences();
          return doamins.contains(QR.params['domain'].toString())
            ? null
            : '/notfound';
       })
     ],
    builder: () =>Dashboard())
```
is this what you looking for?
